docs/farming-&-staking/farming/advanced-cli/cli-farming-cluster.mdx

```diff
@@ -12,137 +13,420 @@ keywords:
+- `/path/to/farm` and `<size>`: The path and size for storing your plots. You can add multiple plots to your farm:
 
-It is possible to execute multiple components from one command line by separating them with **`--`**
+    ```
+    path=/path/to/farm1,size=<size> \
+    path=/path/to/farm2,size=<size>
+    ```
 
+<details>
+<summary>Common Paramaters</summary>
+
+Refer to `./subspace-farmer cluster farmer --help` for an explanation of each option and its default values.
+
+```shell
+--create
 ```
-subspace-farmer cluster --nats-server nats://<NATS_IP>:4222 \
+</details>
+
+</details>
+
+### Grouping Components
+
+<details open>
+<summary>You can run Farming Cluster components individually or combine them to run simultaneously.</summary>
+
+```bash
+    cluster \
+        --nats-server nats://<nats_ip>:4222 \
+    controller \
+        --base-path /path/to/controller-dir \
+        --node-rpc-url ws://<node_ip>:<rpc_port>
+    --
     cache \
-        path=/path/to/cache,size=SIZE
+        path=/path/to/cache,size=<size>
     --
     plotter
+    --
+    farmer \
+        --reward-address <reward_address> \
+        path=/path/to/farm,size=<size>
 ```
 
-### Remote Compute Possibilities
+</details>
+
+---
+
+
+## Advanced Topics
+
+
+### NATS Cluster
 
-This setup allows for extensive remote compute capabilities. For instance, if a computer is suitable for farming but not for plotting, you can run the farmer on that computer and the plotter on one or more remote machines. This enables continuous plotting and replotting across multiple farms, ensuring plotters are utilized efficiently until all sectors are plotted and replotted.
+<details open>
+<summary>Just like every other component in the Farming Cluster, it is possible to have redundent NATS servers. Doing so requires an extra step of configuring those NATS servers in their own cluster.</summary>
 
-### Advanced Options / Redundancy
+Check out the [NATS.io Cluster Guide](https://docs.nats.io/running-a-nats-service/configuration/clustering) for further details.
 
-It is possible to setup clustering with nats server, but that setup is beyond the scope of this document. If you have more than one nats server, you can specify multiple ones.
+<div style={{ textAlign: 'center' }}>
 
+```mermaid
+flowchart
+    A[Farming Cluster Components] <--> B((NATS Server 1))
+    A[Farming Cluster Components] <--> C((NATS Server 2))
 ```
-subspace-farmer cluster --nats-server nats://<NATS_IP#1>:4222 --nats-server nats://<NATS_IP#2>:4222\
+
+</div>
+
+<details>
+<summary>Example</summary>
+
+```bash
+./subspace-farmer cluster \
+    controller \
+        --nats-server=nats://<nats_ip_1>:4222 \
+        --nats-server=nats://<nats_ip_2>:4222 \
+        --base-path /path/to/controller-dir \
+        --node-rpc-url ws://<node_ip>:<rpc_port>        
 ```
 
-You can run multiple controller components, but each one needs to have it’s distinct cache group.
+</details>
+
+</details>
+
+
+### Multiple Components
```

Comment:
I'm wondering if we should put a switch on such diagrams, though conceptually this is accurate

Reply:
I refrained from adding a switch in the diagram to keep it simple. Additionally, not every component necessarily connects to a physical switch, for example if they're using a loopback adapter for Components like NATS, Controller, Cache, and Plotter on a single system.

Here is a comparison of the diagrams. Let me know if you think adding the switch would be beneficial.

![image](https://github.com/user-attachments/assets/fa929725-3ebf-4f60-8094-075ed961c16a)